tests/index.spec.ts

```diff
@@ -1,32 +1,138 @@
 import { readFileSync } from 'fs';
+import { watch } from 'fs/promises';
+import { setTimeout } from 'timers/promises';
+import { Client } from '@elastic/elasticsearch';
 import jsLogger from '../src';
 
+async function waitForFileWrite(path: string): Promise<void> {
+  const watcher = watch('.', { persistent: false });
+
+  for await (const event of watcher) {
+    if (event.filename === path && event.eventType === 'change') {
+      return;
+    }
+  }
+}
+
 describe('jsLogger', function () {
-  it('should initialize the logger without errors', function () {
-    const logger = jsLogger();
+  describe('basic', function () {
+    it('should initialize the logger without errors', function () {
+      const logger = jsLogger();
 
-    expect(logger).toBeDefined();
-    expect(() => logger.info('test')).not.toThrow();
-  });
+      expect(logger).toBeDefined();
+      expect(() => logger.info('test')).not.toThrow();
+    });
+
+    it('should support other destinations', async function () {
+      const path = 'avi.log';
+      const logger = jsLogger({}, { file: { enabled: true, path: 'avi.log' }, console: { enabled: false } });
+
+      const waitForFileWritePromise = waitForFileWrite(path);
+
+      logger.info('avi');
+
+      await waitForFileWritePromise;
+
+      const logLine = JSON.parse(readFileSync(path).toString()) as Record<string, string>;
+
+      expect(logLine).toHaveProperty('msg', 'avi');
+    });
+
+    it('should support base option', async function () {
+      const path = 'avi-base.log';
+      const logger = jsLogger({ base: { key: 'value' } }, { file: { enabled: true, path }, console: { enabled: false } });
+
+      const waitForFileWritePromise = waitForFileWrite(path);
+
+      logger.info('avi');
 
-  it('should support other destinations', function () {
-    const logger = jsLogger({}, 'avi.log');
+      await waitForFileWritePromise;
 
-    logger.info('avi');
+      const logLine = JSON.parse(readFileSync(path).toString()) as Record<string, string>;
 
-    const logLine = JSON.parse(readFileSync('avi.log').toString()) as Record<string, string>;
+      expect(logLine).toHaveProperty('msg', 'avi');
+      expect(logLine).toHaveProperty('key', 'value');
+    });
 
-    expect(logLine).toHaveProperty('msg', 'avi');
+    it('should still initialize the logger with all destinations disabled', function () {
+      const logger = jsLogger({}, { file: { enabled: false }, console: { enabled: false } });
```

Comment:
Isn't `file` destination disabled by default?

Reply:
I wanted to show explicitly that all are disabled for the test.

Reply:
Makes sense.